src/imitation/algorithms/dagger.py

```diff
@@ -237,46 +275,55 @@ class DAggerTrainer:
 
     def __init__(
         self,
-        env: gym.Env,
-        scratch_dir: str,
+        venv: vec_env.VecEnv,
+        scratch_dir: types.AnyPath,
         beta_schedule: Callable[[int], float] = None,
         batch_size: int = 32,
-        **bc_kwargs,
+        bc_kwargs: Optional[dict] = None,
     ):
         """Trainer constructor.
 
         Args:
-            env: environment to train in.
+            venv: Vectorized training environment. Note that when the robot
+                action is randomly injected (in accordance with `beta_schedule`
+                argument), every individual environment will get a robot action
+                simulatenously for that timestep.
             scratch_dir: directory to use to store intermediate training
                 information (e.g. for resuming training).
             beta_schedule: provides a value of `beta` (the probability of taking
                 expert action in any given state) at each round of training. If
                 `None`, then `linear_beta_schedule` will be used instead.
             batch_size: Number of samples in each batch during BC training.
-            **bc_kwargs: additional arguments for constructing the `BC` that
+            bc_kwargs: additional arguments for constructing the `BC` that
                 will be used to train the underlying policy.
         """
         # for pickling
-        self._init_args = locals()
-        self._init_args.update(bc_kwargs)
-        del self._init_args["self"]
-        del self._init_args["bc_kwargs"]
+        # self._init_args = locals()
+        # self._init_args.update(bc_kwargs)
+        # del self._init_args["self"]
+        # del self._init_args["bc_kwargs"]
```

Comment:
I did some experimenting and it turned out that `self._init_args` wasn't necessary for getting the "pickle and load" `test_dagger.py` test to pass. Is this vestigal code, @qxcv?

Reply:
That's right, it's safe to remove. Originally this class serialised itself using the "save all the kwargs and any mutable state" trick. After the Torch port it became possible to just pickle the whole thing, which is what the current code does.